python/datasources/cawp_time.py

```diff
@@ -0,0 +1,654 @@
+                             cawp_place_phrase, cawp_place_phrase)
+
+    # only keep 2 letter code portion
+    place_terms_list = cawp_place_phrase.split(" - ")
+    place_code = place_terms_list[1]
+
+    return place_code
+
+
+def get_consecutive_time_periods(first_year: int = DEFAULT_FIRST_YR, last_year: int = DEFAULT_LAST_YR):
+    """ Generates a list of consecutive time periods in the "YYYY" format
+
+    Parameters:
+        first_year: optional int to start the list; defaults to 1915
+            which is two years before the first woman in US Congress
+        last_year: optional int to be the last element in the list
+            other than the default of 2022
+    Returns:
+        a list of string years (e.g. ["1999", "2000", "2001"]) """
+    return [str(x) for x in list(range(first_year, last_year + 1))]
+
+
+def get_state_level_fips():
+    """ Returns list of 2-letter strings for state and territory fips codes """
+    return STATE_LEVEL_FIPS_LIST
+
+
+def add_aian_api_rows(df):
+    """ Adds new rows for the combined AIAN_API race group """
+
+    # only keep rows with years that will get population
+    target_time_periods = get_consecutive_time_periods(
+        first_year=ACS_FIRST_YR, last_year=ACS_LAST_YR)
+    df_aian_api_rows = df[df[std_col.TIME_PERIOD_COL].isin(
+        target_time_periods)]
+
+    # only keep rows with races to be combined
+    df_aian_api_rows = df_aian_api_rows.loc[
+        df_aian_api_rows[RACE_ETH].isin(
+            AIAN_API_RACES)]
+    df_aian_api_rows = df_aian_api_rows[[std_col.TIME_PERIOD_COL,
+                                         std_col.STATE_FIPS_COL,
+                                         std_col.STATE_NAME_COL,
+                                         std_col.STATE_POSTAL_COL,
+                                         std_col.W_THIS_RACE_CONGRESS_NAMES]]
+
+    df_aian_api_rows = df_aian_api_rows.groupby([std_col.TIME_PERIOD_COL,
+                                                 std_col.STATE_FIPS_COL,
+                                                 std_col.STATE_NAME_COL,
+                                                 std_col.STATE_POSTAL_COL
+                                                 ], as_index=False)[
+        std_col.W_THIS_RACE_CONGRESS_NAMES
+    ].agg(lambda nested_list: [x for list in nested_list for x in list])
+
+    # remove any duplicates if a women was in both of the combined race groups
+    df_aian_api_rows[std_col.W_THIS_RACE_CONGRESS_NAMES] = df_aian_api_rows[std_col.W_THIS_RACE_CONGRESS_NAMES].apply(
+        set).apply(list)
+
+    df_aian_api_rows[std_col.W_THIS_RACE_CONGRESS_COUNT] = df_aian_api_rows[std_col.W_THIS_RACE_CONGRESS_NAMES].apply(
+        lambda list: len(list)).astype(float)
+
+    df_aian_api_rows[RACE_ETH] = "AIAN_API"
+
+    df_aian_api_rows = df_aian_api_rows.reset_index(drop=True)
+
+    # re-merge with this to preserve the non-summed rows like "total_congress_count", etc
+    df_only_api_rows = df.loc[df[RACE_ETH].isin(
```

Comment:
why would you do it with `isin` instead of `==`

Reply:
good call i think originally i assumed i was going to be collecting both races here and then changed later